### PR TITLE
Dataexplorer viewlet

### PR DESCRIPTION
**Pull request by abist**
Initial Data Explorer viewlet implmentation under dev flag. Added tests for DataExplorer viewlwt, and OE already has tests.

### CODE REVIEW

File: src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			}
+		}
+
+		return true;
+	}
+
+
+	private getViewContainer(value: string): ViewContainer {
+		switch (value) {
+			case 'azureResource': return this.viewContainersRegistry.get(VIEWLET_ID);
```
Comment (udeeshagautam) [inline src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts]: This is the one I wanted to talk about. What happens when cmsResource comes - will we need to add a new case here?

File: src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		dataExplorerExtensionPoint.setHandler(extensions => {
+			for (let extension of extensions) {
+				const { value, collector } = extension;
+
+				forEach(value, entry => {
+					if (!this.isValidViewDescriptors(entry.value, collector)) {
+						return;
+					}
+
+					let container = this.getViewContainer(entry.key);
```
Comment (anthonydresser) [inline src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts]: I don't think you need this, this contribution point only applies to the data explorer viewlet, so we can just use the VIEWLET_ID directly

File: src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts (hunk 1 above)
Comment (abist) [reply src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts]: Oh I thought you were talking about something else. We can get rid of this too.

File: src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts (hunk 2 above)
Comment (abist) [reply src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts]: Fixed

**Comment by abist** [comment]
@anthonydresser Agreed, but since we will be getting rid of it in future anyway, it's fine for dev purposes now like you pointed out.

**Pull request merged**